src/main/scala/fr/polytechnique/cmap/cnam/etl/transformers/patients/PatientFilters.scala

```diff
@@ -0,0 +1,63 @@
+package fr.polytechnique.cmap.cnam.etl.transformers.patients
+
+import java.sql.Timestamp
+
+import fr.polytechnique.cmap.cnam.etl.extractors.patients.PatientsConfig
+import fr.polytechnique.cmap.cnam.etl.patients.Patient
+import fr.polytechnique.cmap.cnam.util.functions.makeTS
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.functions.{col, floor, lit, months_between}
+
+class PatientFilters(config: PatientsConfig) extends Serializable {
+
+  /** Filter Patient with config
+   *
+   * @param patients the patient to check.
+   * @return [[Dataset]] of Patient
+   */
+  def transform(patients: Dataset[Patient]): Dataset[Patient] = {
+
+    val ageReferenceDate: Timestamp = Timestamp.valueOf(config.ageReferenceDate.atStartOfDay())
+    val age = floor(months_between(lit(ageReferenceDate), col("birthDate")) / 12)
+    val filterPatientsByAge = age >= config.minAge && age < config.maxAge
+
+    import patients.sparkSession.implicits._
+
+    val filteredPatients = patients
+      .where(filterPatientsByAge)
+      .flatMap(controlGender(config.minGender, config.maxGender))
+      .flatMap(controlDeathDate(config.minYear, config.maxYear))
+
+    filteredPatients
+  }
+
+  /** Returns None if Patient is died before minYear or after maxYear, otherwise returns Some(Patient)
+   *
+   * @param patient the patient to check.
+   * @param minYear the minimum year to control with.
+   * @param maxYear the maximum year to control with.
+   * @return an Option of Patient
+   */
+  protected[patients] def controlDeathDate(minYear: Int, maxYear: Int)(patient: Patient): TraversableOnce[Patient] = {
```

Comment:
This logic make me confused. In your transform method, you filter the patients not in the proper range of age but in this method you return none when a patient is not in the condition.

You need to use the same strategy.